System/Process/Env/OsString.hs

```diff
@@ -0,0 +1,32 @@
+{-# LANGUAGE CPP #-}
+
+#if !MIN_VERSION_filepath(1,5,0)
+-- Silence the warning on System.OsString.Internal.Types import. It exists
+-- (deprecated) in filepath 1.4.100.0 && < 1.5.0.0. For filepath >= 1.5,
```

Comment:
I don't see any deprecation notices here:

* https://hackage.haskell.org/package/filepath-1.4.100.0/docs/System-OsString-Internal-Types.html
* https://hackage.haskell.org/package/filepath-1.4.100.1/docs/System-OsString-Internal-Types.html
* https://hackage.haskell.org/package/filepath-1.4.100.2/docs/System-OsString-Internal-Types.html
* https://hackage.haskell.org/package/filepath-1.4.100.3/docs/System-OsString-Internal-Types.html
* https://hackage.haskell.org/package/filepath-1.4.100.4/docs/System-OsString-Internal-Types.html

and compiling with `cabal repl -b filepath==1.4.100.0` and this CPP removed does not emit a warning.

And in any case, I'm personally not fan of using CPP to suppress warnings. Do we really need this CPP?

Reply:
Ah, it's in the 300+ series e.g. https://hackage.haskell.org/package/filepath-1.4.301.0/docs/System-OsString-Internal-Types.html.

In any case, no, it's not necessary. Removed.

Reply:
This doesn't seem to have been removed. Has a push gone astray?